enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1185,22 +1185,24 @@ func (p *PebbleCache) Writer(ctx context.Context, r *resource.ResourceName) (int
 			return err
 		}
 
+		sizeDelta := bytesWritten
 		iter := db.NewIter(nil /*default iterOptions*/)
 		defer iter.Close()
-		alreadyExists := pebbleutil.IterHasKey(iter, fileMetadataKey)
-		if alreadyExists {
+		existingMD, err := lookupFileMetadata(ctx, iter, fileMetadataKey)
+		if err == nil {
 			metrics.DiskCacheDuplicateWrites.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Inc()
 			metrics.DiskCacheDuplicateWritesBytes.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Add(float64(r.GetDigest().GetSizeBytes()))
 			tracing.AddStringAttributeToCurrentSpan(ctx, "pebble.duplicate_write", "true")
+			sizeDelta = bytesWritten - existingMD.GetStoredSizeBytes()
 		}
 
 		err = db.Set(fileMetadataKey, protoBytes, &pebble.WriteOptions{Sync: false})
 		if err == nil {
 			partitionID := fileRecord.GetIsolation().GetPartitionId()
-			if !alreadyExists {
+			if sizeDelta != 0 {
 				p.sendSizeUpdate(partitionID, fileMetadataKey, bytesWritten)
 			}
-			metrics.DiskCacheAddedFileSizeBytes.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Observe(float64(bytesWritten))
+			metrics.DiskCacheAddedFileSizeBytes.With(prometheus.Labels{metrics.CacheNameLabel: p.name}).Observe(float64(sizeDelta))
```

Comment:
Is it intended that this metric could have a negative value? Not sure whether this metric would be more useful if we used sizeDelta to show that we are overwriting files to be smaller, or if we used bytesWritten if we wanted to know the actual sizes of files on the disk

Reply:
This is supposed to be bytesWritten. I changed the wrong line :facepalm: